pkg/xds/envoy/routes/v2/routes_configurer.go

```diff
@@ -148,5 +182,34 @@ func (c RoutesConfigurer) routeAction(clusters []envoy_common.Cluster) *envoy_ro
 			AutoHostRewrite: &wrappers.BoolValue{Value: true},
 		}
 	}
-	return &routeAction
+	c.setModifications(routeAction, modify)
+	return routeAction
+}
+
+func (c RoutesConfigurer) setModifications(routeAction *envoy_route.RouteAction, modify *mesh_proto.TrafficRoute_Http_Modify) {
+	if modify.GetPath() != nil {
+		switch modify.GetPath().Type.(type) {
+		case *mesh_proto.TrafficRoute_Http_Modify_Path_RewritePrefix:
+			routeAction.PrefixRewrite = modify.GetPath().GetRewritePrefix()
+		case *mesh_proto.TrafficRoute_Http_Modify_Path_Regex:
+			routeAction.RegexRewrite = &envoy_type_matcher.RegexMatchAndSubstitute{
+				Pattern: &envoy_type_matcher.RegexMatcher{
+					EngineType: &envoy_type_matcher.RegexMatcher_GoogleRe2{
+						GoogleRe2: &envoy_type_matcher.RegexMatcher_GoogleRE2{},
```

Comment:
Do we still need to deal with the max RE2 program size here? Docs say that even though its deprecated, the default is still 100 and I don't see this being set in the layered runtime.

Reply:
Looks like the default is still 100. I think it makes sense to address this in a separate PR before the next release #2103